api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -260,9 +260,15 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
 
               queryJobConfiguration = queryJobConfiguration.toBuilder().setQuery(query).build();
 
+              /* Google appengine has a 60 second timeout, we want to make sure this endpoint completes
+               * before that limit is exceeded, or we get a 500 error with the following type:
+               * com.google.apphosting.runtime.HardDeadlineExceededError
+               * See https://cloud.google.com/appengine/articles/deadlineexceedederrors for details
+               */
               TableResult queryResponse =
                   bigQueryService.executeQuery(
-                      bigQueryService.filterBigQueryConfig(queryJobConfiguration), 60000l);
+                      bigQueryService.filterBigQueryConfig(queryJobConfiguration),
+                      50000l / retryLimit + 1);
```

Comment:
Can you convert retryLimit to a class-level constant with ALL_CAPS? Right now it's statically defined at the start of this function and never reassigned – would be easier to read if it's clearly a constant.

(By the same token – 50000l is a bit of a magic number, which could also be a constant APP_ENGINE_HARD_TIMEOUT_MSEC or something

Reply:
Will do. This bug was occuring relatively rarely, but happened more often with large value sets, and with all participants. It is hard to reliably repro locally, because we aren't running an actual appengine server, so we don't get this error.